http-service/src/main/java/net/runelite/http/service/adventurelog/AdventureLogController.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2019, Trevor <https://github.com/Trevor159>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.http.service.adventurelog;
+
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import net.runelite.http.service.account.AuthFilter;
+import net.runelite.http.service.account.beans.SessionEntry;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import static org.springframework.web.bind.annotation.RequestMethod.POST;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/adventurelog")
+public class AdventureLogController
+{
+	@Autowired
+	private AdventureLogService adventureLogService;
+
+	@Autowired
+	private AuthFilter authFilter;
+
+	@GetMapping
+	public List<Object> getLogs(
+		HttpServletRequest request,
+		HttpServletResponse response,
+		@RequestParam String username) throws IOException
+	{
+		SessionEntry session = authFilter.handle(request, response);
+
+		if (session == null)
+		{
+			return null;
+		}
+
+		return adventureLogService.getLogs(session.getUser(), username);
+	}
+
+	@RequestMapping(method = POST)
+	public void addLog(
+		HttpServletRequest request,
+		HttpServletResponse response,
+		@RequestParam String username,
+		@RequestBody Object values
```

Comment:
http service has dependency on http api, what is the point using Object as body when you can simply use AdventureLogRecord collection?

Reply:
Mongo doesn't seem to know how to parse POJOs like logData. It parses general Objects fine so I just kept it as an Object.

Reply:
Then just serialize it by yourself